compiler/src/dotty/tools/dotc/typer/Applications.scala

```diff
@@ -768,13 +812,41 @@ trait Applications extends Compatibility {
     def addArg(arg: Tree, formal: Type): Unit =
       typedArgBuf += adapt(arg, formal.widenExpr)
 
+
     def makeVarArg(n: Int, elemFormal: Type): Unit = {
       val args = typedArgBuf.takeRight(n).toList
       typedArgBuf.dropRightInPlace(n)
       val elemtpt = TypeTree(elemFormal)
-      typedArgBuf += seqToRepeated(SeqLiteral(args, elemtpt))
+      val y = seqToRepeated(SeqLiteral(args, elemtpt))
+      // println(s"form once arguments are added together to the buffer: ${y}")
+      typedArgBuf += y
+    }
+
+    def makeSeqLiteral(n: Int, elemFormal: Type): Unit = {
+      val args = typedArgBuf.takeRight(n).toList
+      typedArgBuf.dropRightInPlace(n)
+      val elemtpt = TypeTree(elemFormal) 
+      typedArgBuf += SeqLiteral(args, elemtpt)
     }
 
+    def argCombineVarArg(n: Int, elemFormal: Type): Unit =                       
+      val varArgTrees = typedArgBuf.takeRight(n)
+      typedArgBuf.dropRightInPlace(n)
+      val repeatedarg = seqToRepeated(
+                          varArgTrees.reduce{
+                            (lh, rh) => 
+                              if ((lh.tpe.isRepeatedParam) && (rh.tpe.isRepeatedParam)) then 
+                                  (lh.asInstanceOf[Typed].expr).select(defn.Seq_++).appliedToType(elemFormal).appliedTo(rh.asInstanceOf[Typed].expr)
+                              else if ( (lh.tpe.isRepeatedParam) ) then 
+                                  (lh.asInstanceOf[Typed].expr).select(defn.Seq_++).appliedToType(elemFormal).appliedTo(rh)
+                              else if ( (rh.tpe.isRepeatedParam) ) then 
+                                  (lh).select(defn.Seq_++).appliedToType(elemFormal).appliedTo(rh.asInstanceOf[Typed].expr)
+                              else lh.select(defn.Seq_++).appliedToType(elemFormal).appliedTo(rh)})
```

Comment:
you can simplify this by defining a method that takes a `Tree[Type]` and returns the same tree without the `*` if there is a `*`.

In the end you should just have:
```scala
removeRepeated(lh).appliedTo(elemFormal).appliedTo(removeRepeated(rh))
```